include/seqan/index/index_bifm.h

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2006-2015, Knut Reinert, FU Berlin
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Christopher Pockrandt <christopher.pockrandt@fu-berlin.de>
+// ==========================================================================
+
+//SEQAN_NO_DDDOC:do not generate documentation for this file
+
+#ifndef INDEX_BIFM_H_
+#define INDEX_BIFM_H_
+
+namespace seqan {
+
+// ==========================================================================
+// Forwards
+// ==========================================================================
+
+template <typename TSpec = void, typename TConfig = FMIndexConfig<TSpec> >
+class BidirectionalFMIndex;
+
+// ============================================================================
+// Metafunctions
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Metafunction BiFMReversedText
+// ----------------------------------------------------------------------------
+
+/*!
+ * @class BiFMReversedText
+ * @headerfile <seqan/index.h>
+ * @brief A helper object that stores the type of the modifier of a given type @endlink.
+ *
+ * @signature template <TText>
+ *            struct BiFMReversedText;
+ */
+
+template <typename TText>
+struct BiFMReversedText
```

Comment:
This ```BiFMReversedText ``` metafunction would be replaced by a ```RevTextFibre```, or maybe even get removed if fwd/rev indices work on the same (fwd) text type.